crates/cast/src/helpers/scarb_utils.rs

```diff
@@ -38,11 +39,31 @@ pub fn get_property(
     }
 }
 
+pub fn get_scarb_manifest(path: &Option<Utf8PathBuf>) -> Result<Utf8PathBuf> {
+    which::which("scarb")
+        .context("Cannot find `scarb` binary in PATH. Make sure you have Scarb installed https://github.com/software-mansion/scarb")?;
+    path.as_ref().map_or_else(
+        || {
+            let output = Command::new("scarb")
+                .arg("manifest-path")
+                .stdout(Stdio::piped())
+                .output()
+                .expect("Failed to execute scarb manifest-path command");
+            let output_str = String::from_utf8(output.stdout)
+                .expect("Invalid output of scarb manifest-path command");
+            let path = Utf8PathBuf::from_str(output_str.trim())
+                .expect("Scarb manifest-path returned invalid path");
```

Comment:
We would like to improve the errors and I wonder if we should panic here or print something like

```
command: ...
error: Failed to execute scarb manifest-path command
```

Reply:
You can improve added `expects` here and we should also create an issue for the rest

Reply:
Changed in [commit](https://github.com/foundry-rs/starknet-foundry/pull/340/commits/9a6634311907e0812c7910151045d3c5b3ca4fa1). Created an issue https://github.com/foundry-rs/starknet-foundry/issues/352